core/market/decentralized/src/testing.rs

```diff
@@ -21,3 +21,15 @@ pub mod proposal_util;
 
 pub use mock_node::{wait_for_bcast, MarketServiceExt, MarketsNetwork};
 pub use mock_offer::{client, sample_demand, sample_offer};
+
+pub fn generate_backtraced_name() -> String {
+    let bt = backtrace::Backtrace::new();
+    // 0th element should be this function. We'd like to know the caller
+    let frame = &bt.frames()[1];
+    for symbol in frame.symbols().iter() {
+        if let Some(name) = symbol.name() {
+            return name.to_string().to_string();
+        }
+    }
+    uuid::Uuid::new_v4().to_string().to_string()
```

Comment:
cool that there is the fallback, but maybe it is better to see it failing, to understand how it works in long term?